File: Melodex-UserSphere/src/Validator.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Validator {

    private final Pattern pattern;

    public Validator(String regex) {
        this.pattern = Pattern.compile(regex);
    }

    public boolean checkRegex(String message) {
        if (message == null) {
            throw new NullPointerException("Message cannot be null");
        }
        Matcher matcher = pattern.matcher(message);
        return matcher.matches();
    }
}

File: internal/validator/validator.go
package validator

import (
	"fmt"
	"regexp"
)

type Validator struct {
	Regex string
}

func (v *Validator) CheckInput(message string) (bool, error) {
	re, err := regexp.Compile(v.Regex)
	if err != nil {
		return false, fmt.Errorf("invalid regex pattern: %v", err)
	}
	match := re.MatchString(message)
	return match, nil
}
